helix-core/src/main/java/org/apache/helix/task/JobDispatcher.java

```diff
@@ -85,15 +86,15 @@ public ResourceAssignment processJobStatusUpdateAndAssignment(String jobName,
     // completed)
     TaskState workflowState = workflowCtx.getWorkflowState();
     TaskState jobState = workflowCtx.getJobState(jobName);
-    // The job is already in a final state (completed/failed).
+    // Do not include workflowState == TIMED_OUT here, as later logic needs to handle this case
```

Comment:
I am assuming this is not related to this PR? Anyways it is fine to have this comment, I just want to make sure I understand it correctly.

Reply:
No it's not related to this PR. I added it as I realized why we shouldn't have TIMED_OUT here, and other devs should know too.